src/openrct2/entity/EntityTweener.cpp

```diff
@@ -10,121 +10,176 @@
 
 #include "../entity/Guest.h"
 #include "../entity/Staff.h"
+#include "../interface/Viewport.h"
+#include "../interface/Window.h"
 #include "../ride/Vehicle.h"
 #include "EntityList.h"
 #include "EntityRegistry.h"
 
 #include <algorithm>
 #include <cmath>
 
-void EntityTweener::AddEntity(EntityBase* entity)
+namespace OpenRCT2
 {
-    Entities.push_back(entity);
-    PrePos.emplace_back(entity->GetLocation());
-}
-
-void EntityTweener::PopulateEntities()
-{
-    for (auto ent : EntityList<Guest>())
+    static inline ViewportList GetUnzoomedViewports() noexcept
     {
-        AddEntity(ent);
+        ViewportList viewports;
+        WindowVisitEach([&](WindowBase* w) {
+            if (auto* vp = WindowGetViewport(w); vp != nullptr)
+            {
+                // TODO: Check if viewport is visible, depends on PR #24464
```

Comment:
Should we wait for #24464 to be merged first?

Reply:
Don't mind it, I believe the PR is ready to go anyway.